lib/semantics/tools.h

```diff
@@ -84,6 +84,10 @@ bool CanBeTypeBoundProc(const Symbol *);
 const Symbol *FindUltimateComponent(
     const DerivedTypeSpec &type, std::function<bool(const Symbol &)> predicate);
 
+// Returns an immediate component of type that matches predicate, or nullptr.
+const Symbol *FindImmediateComponent(
+    const DerivedTypeSpec &, const std::function<bool(const Symbol &)> &);
```

Comment:
It might be helpful to define "immediate component" in the comment below that defined "ultimate component", etc.

Reply:
Will do.